etna/transforms/decomposition/model_based.py

```diff
@@ -0,0 +1,185 @@
+from typing import List
+from typing import Union
+from typing import get_args
+
+import pandas as pd
+
+from etna.datasets import TSDataset
+from etna.datasets.utils import determine_num_steps
+from etna.models import BATSModel
+from etna.models import DeadlineMovingAverageModel
+from etna.models import HoltWintersModel
+from etna.models import ProphetModel
+from etna.models import SARIMAXModel
+from etna.models import SeasonalMovingAverageModel
+from etna.models import TBATSModel
+from etna.models.base import ContextRequiredModelType
+from etna.models.base import ModelType
+from etna.transforms import IrreversibleTransform
+
+_SUPPORTED_MODELS = Union[
+    HoltWintersModel,  # full
+    ProphetModel,  # full
+    SARIMAXModel,  # full
+    DeadlineMovingAverageModel,  # need to account context/prediction size
+    SeasonalMovingAverageModel,  # need to account context/prediction size
+    BATSModel,  # dynamic components, not reliable
```

Comment:
What does it mean "not reliable"?

Reply:
As I understood, this will result in a warning.

Reply:
Yes, and it is may be a bit confusing if warnings are suppressed or not paid enough attention